### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by nedjwab**
### In This pull request i :
- Create an app. rb file that will serve as a console app entry-point. It should have methods that do the following:

> List all books.
> List all people.
> Create a person (teacher or student, not a plain Person).
> Create a book.
> Create a rental.
> List all rentals for a given person's id

-In the main.rb define the entry point, this will be a method called main that is invoked at the end of the file. This method should do the following:

> - Present the user with a list of options to perform.
> - Lets users choose an option.
> - If needed, ask for parameters for the option.
> - Have a way to quit the app.

- Fix RoboCup errors.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    when 'N', 'NO'
+      permission = false
+    end
+    @people.push(Student.new(nil, age, name, parent_permission: permission))
+    puts 'Student created successfuly!'
+  end
+
+  def create_rental
+    fetch_books
+    print 'Select the key of the book: '
```
Comment (BenMukebo) [inline app.rb]: - [ ] Your app.rb file that will serve as their console app entry-point should have methods that do and look something like the following:


```ruby
# *Create a person (teacher or student, not a plain `Person`)
  def create_person
    print "Do you want to create a student (1) or a teacher (2)? [Input the number]: "
    # *If needed, ask for parameters for the option.
    person_type = gets.chomp
    if person_type != "1" && person_type != "2"
      puts "Invalid option"
      return
    end

    print "Age: "
    # *If needed, ask for parameters for the option.
    age = gets.chomp

    print "Name: "
    # *If needed, ask for parameters for the option.
    name = gets.chomp

    person =
      if person_type == "1"
        print "Has parent permission? [Y/N]: "
        # *If needed, ask for parameters for the option.
        parent_permission = gets.chomp
        parent_permission = parent_permission.downcase == "y"

        Student.new(age, name, parent_permission)
      elsif person_type == "2"
        print "Specialization: "
        # *If needed, ask for parameters for the option.
        specialization = gets.chomp

        Teacher.new(age, specialization, name)
      end

    @people << person
    puts "Person created successfully"
  end

  # *Create a book.
  def create_book
    print "Title: "
    # *If needed, ask for parameters for the option.
    title = gets.chomp

    print "Author: "
    # *If needed, ask for parameters for the option.
    author = gets.chomp

    @books << Book.new(title, author)
    puts "Book created successfully"
  end
  
  # *Create a rental.
  def create_rental
    puts "Select a book from the following list by number"
    @books.each_with_index do |book, index|
      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
    end
    # *If needed, ask for parameters for the option.
    book_index = gets.chomp.to_i

    puts "\nSelect a person from the following list by number (not id)"
    @people.each_with_index do |person, index|
      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
    end
    # *If needed, ask for parameters for the option.
    person_index = gets.chomp.to_i

    print "\nDate: "
    # *If needed, ask for parameters for the option.
    date = gets.chomp

    @rentals << Rental.new(date, @books[book_index], @people[person_index])
    puts "Rental created successfully"
  end
  
```

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require './app'
+
+def main
```
Comment (BenMukebo) [inline main.rb:3]: - [ ] According to the project requirements, the school library console app should behave in the same way as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) it was provided for them. Actually, your console does not work as expected as you can see on the screen video 👇🏻 

[Screencast from 08-05-2022 02:19:07 PM.webm](https://user-images.githubusercontent.com/79058364/183076536-9fc5de46-a2c1-4adc-ace0-67f50d9b5880.webm)

- Kindly address this!!

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @nedjwab,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
# Status: Approved :tada: :tada: 




Hi @nedjwab , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**